src/hal/machine.cpp

```diff
@@ -89,6 +89,18 @@ namespace os::detail {
     MINFO("Initializing heap\n");
     auto main_mem = memory().allocate_largest();
     MINFO("Main memory detected as %zu b\n", main_mem.size);
+    memory().deallocate(main_mem.ptr, main_mem.size);
+
+#ifndef PLATFORM_x86_solo5
+    static const size_t SYSTEMLOG_SIZE = 65536;
+    const size_t LIU_SIZE = (main_mem.size / (100 / 25)) & ~0xfff;
```

Comment:
the last part is to make sure its rounded down to 4k ? so you get somewhat less than 25% of all ram